### PR TITLE
MULTIARCH-4696: Manage the creation of VPC connections to the Transit Gateway

### PR DESCRIPTION
**Pull request by mpkredhat**
MULTIARCH 4696 : Deleting VPC Connections for Transit Gateway and recreate ( Fix for 29 Nov )

### CODE REVIEW

**Comment by prb112** [comment]
/title MULTIARCH-4696 : Manage the creation of VPC connections to the Transit Gateway

**Comment by prb112** [comment]
/retitle [MULTIARCH-4696](https://issues.redhat.com//browse/MULTIARCH-4696) : Manage the creation of VPC connections to the Transit Gateway

**Comment by prb112** [comment]
/retitle MULTIARCH-4696: Manage the creation of VPC connections to the Transit Gateway

**Comment by mpkredhat** [comment]
Code review comments added from Old PR now. 
Old PR is closed now. 
This is continuation of PR#[59296](https://github.com/openshift/release/pull/59296)
/cc @prb112

File: ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh [hunk 1]
```diff
@@ -24,39 +24,22 @@ function cleanup_ibmcloud_powervs() {
...
+      ic tg connection-delete "${GW}" "${CS}" --force || true
+      sleep 120
+      echo "Done Cleaning up GW VPC Connection"
+    else
+      echo "GW VPC Connection not found. VPC Cleanup not needed."
     fi
+    break
   done
-
+
```
Comment (prb112) [inline ci-operator/step-registry/upi/deprovision/heterogeneous/powervs/upi-deprovision-heterogeneous-powervs-commands.sh:42]:
```suggestion

```

File: ci-operator/step-registry/upi/install/heterogeneous/powervs/upi-install-heterogeneous-powervs-commands.sh [hunk 2]
```diff
@@ -377,6 +360,18 @@ case "$CLUSTER_TYPE" in
...
 
+      #Create the VPC Connection for the TG
+      for GW in $(ic tg gateways --output json | jq --arg resource_group "${RESOURCE_GROUP_ID}" --arg workspace_name "${WORKSPACE_NAME}" -r '.[] | select(.resource_group.id == $resource_group) | select(.name == $workspace_name) | "(.id)"')
+      do
+	      for CS in $(ic is vpcs --output json | jq -r '.[] | select(.name | contains("${WORKSPACE_NAME}-vpc")) | .id')
+	      do
+  	      VPC_CONN_NAME=$(ic is vpc "${CS}" --output json | jq -r .name)
+	        VPC_NW_ID=$(ic is vpc "${CS}" --output json | jq -r .crn)
+	        echo "Creating new VPC connection for gateway now."
+	        ic tg cc "${GW}" --name "${VPC_CONN_NAME}" --network-id "${VPC_NW_ID}" --network-type vpc || true
```
Comment (prb112) [inline ci-operator/step-registry/upi/install/heterogeneous/powervs/upi-install-heterogeneous-powervs-commands.sh:371]:
```suggestion
  	        VPC_CONN_NAME=$(ic is vpc "${CS}" --output json | jq -r .name)
	        VPC_NW_ID=$(ic is vpc "${CS}" --output json | jq -r .crn)
	        echo "Creating new VPC connection for gateway now."
	        ic tg cc "${GW}" --name "${VPC_CONN_NAME}" --network-id "${VPC_NW_ID}" --network-type vpc || true
```

**Pull request merged**